edgepi-dac.js

```diff
@@ -0,0 +1,72 @@
+const { checkValidWriteInput } = require('./helpers/nodeHelpers');
+
+module.exports = function (RED) {
+    const rpc = require('@edgepi-cloud/edgepi-rpc')
+  
+    function DacNode(config) {
+      // Create new node instance with user config
+      RED.nodes.createNode(this, config);
+      const node = this;
+      const ipc_transport = "ipc:///tmp/edgepi.pipe"
+      const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
+      const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
+      const voltage = parseFloat(config.voltage);
+      const channel = rpc.DACChannel[config.dacChannel];
+
+      // Init dac
+      const dac = new rpc.DacService(transport)
+  
+      if (dac){
+        console.info("DAC node initialized on:", transport);
+        node.status({fill:"green", shape:"ring", text:"dac initialized"});
+      }
+
+      // Enforce gain on write method and check valid configuration input
+      let validWriteInput;
+      if(config.method === "write"){
+        dac.setDacGain(true,true);
+        validWriteInput = checkValidWriteInput(voltage, node);
+      }
```

Comment:
if above if statement fails( DAC node not initialized or new rpc.DacService(trasnport) fails), wouldn't this automatically fail?

Reply:
yea. how's this? -> db61ae79a3d9f0e69fff67112bf863ae0fb56eba